src/main/java/duke/domain/Task.java

```diff
@@ -0,0 +1,120 @@
+package duke.domain;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import duke.shared.StringHelpers;
+
+/**
+ * Encapsulates a generic task.
+ */
+public class Task {
+
+    public static final String TYPE_STRING = null;
+    private String typeString = null;
+    private TaskState state;
+    private String name;
+
+
+    static enum TaskState {
+        NEW(" ", "0"), DONE("X", "1");
+
+        private String representation;
+        private String storedRepresentation;
+
+        private TaskState(String representation, String storedRepresentation) {
+            this.representation = representation;
+            this.storedRepresentation = storedRepresentation;
+        }
+
+        public String getStoredRepresentation() {
+            return this.storedRepresentation;
+        }
+
+        public String toString() {
+            return StringHelpers.bracketWrap(representation);
+        }
+    }
+
+
+    /**
+     * Creates an incomplete task.
+     *
+     * @param name Name of task.
+     */
+    public Task(String name) {
+        try {
+            state = TaskState.NEW;
+            this.setName(name);
+            typeString = (String) this.getClass().getField("TYPE_STRING").get(null);
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            System.out.println("Reflection error");
```

Comment:
Good idea! I'm still going to avoid using the PrintableMixin here though because I prefer implementing fewer interfaces (or mixins in this case) if the use case is not major. However, I will definitely keep this in mind if I expand the use of print statements.